pages/games/[hash].js

```diff
@@ -1,34 +1,80 @@
-import styles from "../../styles/Game.module.css";
+import { useState } from "react";
 
 function Game({ data }) {
+  const PLAYER = "PLAYER";
+  const OPPONENT = "OPPONENT";
+
+  const [visibleBoard, setVisibleBoard] = useState(PLAYER);
+
+  function toggleBoard() {
+    if (visibleBoard == PLAYER) {
+      setVisibleBoard(OPPONENT);
+    } else {
+      setVisibleBoard(PLAYER);
+    }
+  }
+
   return (
-    <div>
-      Game page {data.game_hash}
-      <div className={styles.board}>
-        <div className={styles.fleet}>
-          {data.ships.map((ship, i) => (
-            <div
-              key={ship.id}
-              className={styles.square}
-              style={formatGridAttributes(ship.grid_attributes)}
-            >
-              <div className={styles.ship}>
-              </div>
+    <main className="w-screen p-5 flex flex-col">
+      <h1 className="text-2xl">Game page {data.game_hash}</h1>
+
+      <div className="flex items-center justify-between my-2">
+        {visibleBoard == PLAYER ? <h2>Ma flotte</h2>:<h2>Flotte Adverse</h2>}
+        
+        <button onClick={() => toggleBoard()} className="btn-blue">
+          {visibleBoard == PLAYER ? 'Voir la flotte Adverse':'Voir ma flotte'}
+        </button>
+      </div>
+
+      <div className="relative pb-full">
+        {visibleBoard == PLAYER ? (
+          <>
+            
+            <div className="absolute grid grid-cols-10 grid-rows-10 w-full h-full gap-px max-w-lg max-h-lg">
+              {createSquares().map((square, i) => (
+                <div key={i} className="w-full h-full bg-blue-500"></div>
+              ))}
             </div>
-          ))}
-        </div>
+            <div className="absolute grid grid-cols-10 grid-rows-10 w-full h-full gap-px max-w-lg max-h-lg">
+              {data.ships.map((ship, i) => (
+                <div
+                  key={i}
+                  className="p-1"
+                  style={formatGridAttributes(ship.grid_attributes)}
+                >
+                  <div className="w-full h-full bg-green-500 rounded-full"></div>
+                </div>
+              ))}
+            </div>
+          </>
+        ) : (
+          <>
+            <div className="absolute grid grid-cols-10 grid-rows-10 w-full h-full gap-px max-w-lg max-h-lg">
+              {createSquares().map((square, i) => (
+                <div key={i} className="w-full h-full bg-blue-500"></div>
+              ))}
+            </div>
+          </>
+        )}
       </div>
-    </div>
+    </main>
   );
 }
 
 export async function getServerSideProps(context) {
-  const res = await fetch(`${process.env.API_HOST}/games/${context.params.hash}`);
+  const res = await fetch(`${process.env.NEXT_PUBLIC_API_HOST}/games/${context.params.hash}`);
   const data = await res.json();
-
   return { props: { data } };
 }
 
+const createSquares = () => {
```

Comment:
You don't need a method here. And you can use a shorter version `[...Array(100).keys()]` or `Array.from(Array(100).keys())`.

Reply:
Searched something like that but without success. Thank you !

Reply:
Personally I use ` Array.from({length: 100})` (more in https://2ality.com/2018/12/creating-arrays.html)

Reply:
@zyhou It's just for readability or for performances too?